pages/forms/TripForm/[id].js

```diff
@@ -0,0 +1,83 @@
+import CancelButton from "@/components/CancelButton";
+import CreateButton from "@/components/CreateButton";
+import { useRouter } from "next/router";
+import useSWR from "swr";
+
+const fetcher = (...args) => fetch(...args).then((res) => res.json());
+
+export default function TripDetailsForm() {
+  const { mutate } = useSWR("/api/trips");
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: trips, isLoading } = useSWR(
+    id ? `/api/trips/${id}` : null,
+    fetcher
+  );
+
+  if (!trips || isLoading) {
+    return <>is Loading...</>;
+  }
+  const { title, location, timePeriod, img } = trips;
+  const country = location.map((location) => `${location.country}`);
+  const city = location.map((location) => `${location.city}`);
+  const startDate = timePeriod.map((timePeriod) => `${timePeriod.startDate}`);
+  const endDate = timePeriod.map((timePeriod) => `${timePeriod.endDate}`);
+
+  async function handleEditTrip(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const tripData = Object.fromEntries(formData);
+
+    const editedTrip = {
+      title: tripData.title,
+      location: [{ country: tripData.country, city: tripData.city }],
+      timePeriod: [
+        { startDate: tripData.startDate, endDate: tripData.endDate },
+      ],
+    };
+
+    const response = await fetch(`/api/trips/${id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(editedTrip),
+    });
+
+    if (!response.ok) {
+      console.error(response.status);
+      return;
+    }
+
+    mutate();
+    router.push(`/trip/${id}`);
+  }
```

Comment:
If you want to ensure you have proper error handling for your asynchronous functions to prevent unexpected issues, please see my "dry-written" (untested) code snippet below.  😁
_Side note: his could also be a task for a refactor story._

```js
 try {
      const response = await fetch(`/api/trips/${id}`, {
        method: "PUT",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(editedTrip),
      });

      if (!response.ok) {
        throw new Error(`HTTP error! Status: ${response.status}`);
      }

      mutate();
      router.push(`/trip/${id}`);
    } catch (error) {
      console.error("Error updating trip:", error);
    }
```